plumbing/transport/ssh/auth_method_test.go

```diff
@@ -94,18 +94,9 @@ func (s *SuiteCommon) TestPublicKeysCallbackString(c *C) {
 	c.Assert(a.String(), Equals, fmt.Sprintf("user: test, name: %s", PublicKeysCallbackName))
 }
 func (s *SuiteCommon) TestNewSSHAgentAuth(c *C) {
-	addr := os.Getenv("SSH_AUTH_SOCK")
-	err := os.Unsetenv("SSH_AUTH_SOCK")
+	auth, err := NewSSHAgentAuth("foo")
```

Comment:
This test should be skipped if SSH_AUTH_SOCK is not defined:
```
os.Getenv("SSH_AUTH_SOCK") == "" {
                c.Skip("SSH_AUTH_SOCK or SSH_TEST_PRIVATE_KEY are required")
                return
        }
```

Also, the old test tested that the right error was returned when agent is not running, please, preserve that test case.